pyabc/storage/history.py

```diff
@@ -369,48 +360,47 @@ def get_all_populations(self):
 
     @with_session
     @internal_docstring_warning
-    def store_initial_data(self, ground_truth_model: int, options: dict,
+    def store_initial_data(self,
+                           ground_truth_model: int,
+                           options: dict,
                            observed_summary_statistics: dict,
                            ground_truth_parameter: dict,
                            model_names: List[str],
                            distance_function_json_str: str,
                            eps_function_json_str: str,
-                           population_strategy_json_str: str):
+                           population_strategy_json_str: str,
+                           start_time: datetime.datetime = None) -> None:
         """
         Store the initial configuration data.
 
         Parameters
         ----------
-
-        ground_truth_model: int
+        ground_truth_model:
             Index of the ground truth model.
-
-        options: dict
+        options:
             Of ABC metadata.
-
-        observed_summary_statistics: dict
+        observed_summary_statistics:
             The measured summary statistics.
-
-        ground_truth_parameter: dict
+        ground_truth_parameter:
             The ground truth parameters.
-
-        model_names: List
+        model_names:
             A list of model names.
-
-        distance_function_json_str: str
+        distance_function_json_str:
             The distance function represented as json string.
-
-        eps_function_json_str: str
+        eps_function_json_str:
             The epsilon represented as json string.
-
-        population_strategy_json_str: str
+        population_strategy_json_str:
             The population strategy represented as json string.
-
+        start_time:
+            Start time of the analysis.
         """
+        if start_time is None:
```

Comment:
why do we need this check? Does that mean the user can define the start time? if yes, why?

Reply:
valid question. The user cannot define it, but I thought it is more transparent if that time is defined in the ABCSMC class, and not implicitly in the History class. Because then whoever changes the code knows that putting that function to another place may affect the time that is saved in the history.